core/timed/taskexecutor.go

```diff
@@ -0,0 +1,90 @@
+package timed
+
+import (
+	"sync"
+	"time"
+
+	"github.com/iotaledger/hive.go/core/generics/shrinkingmap"
+)
+
+// region TaskExecutor ////////////////////////////////////////////////////////////////////////////////////////////
+
+// TaskExecutor is a TimedExecutor that internally manages the scheduled callbacks as tasks with a unique
+// identifier. It allows to replace existing scheduled tasks and cancel them using the same identifier.
+type TaskExecutor[T comparable] struct {
+	*Executor
+	queuedElements      *shrinkingmap.ShrinkingMap[T, *QueueElement]
+	queuedElementsMutex sync.Mutex
+}
+
+// NewTaskExecutor is the constructor of the TaskExecutor.
+func NewTaskExecutor[T comparable](workerCount int) *TaskExecutor[T] {
+	return &TaskExecutor[T]{
+		Executor:       NewExecutor(workerCount),
+		queuedElements: shrinkingmap.New[T, *QueueElement](),
+	}
+}
+
+// ExecuteAfter executes the given function after the given delay.
+func (t *TaskExecutor[T]) ExecuteAfter(identifier T, callback func(), delay time.Duration) *ScheduledTask {
```

Comment:
Isn't that just exactly the same as `ExecuteAt` with `time.Now().Add(delay)` ? So we can get rid of the duplicate code